src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -126,14 +89,17 @@ def terminate
   private
 
   def self.getManifests(env)
-    manifests = []
-    if env.manifest != Puppet::Node::Environment::NO_MANIFEST
-      if File.directory?(env.manifest)
-        manifests = Dir.glob(File.join(env.manifest, '**/*.pp')).sort
-      else
-        manifests = [env.manifest]
+    manifests =
+      case
+        when env.manifest == Puppet::Node::Environment::NO_MANIFEST
+          []
+        when File.directory?(env.manifest)
+          Dir.glob(File.join(env.manifest, '**/*.pp'))
```

Comment:
Did we decide that this glob matches the current `spec` of what files Puppet will look at?  Or does the spec that Henrik wrote up as docs have any more specific restrictions than this?